### ISSUE TITLE
Adding pooling layers

### ISSUE THREAD

**Issue by Om-Pandey** [tensorflow/addons#727]
**Describe the feature and the current behavior/state.**
 It will be a feature needed for downscaling images. Unlike max pooling, min pooling takes the values in the spatial locality of the filter which is the least and would come in handy for white background images, where data to be extracted has comparatively lesser values.

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=7533046 and https://papers.nips.cc/paper/4204-dynamic-pooling-and-unfolding-recursive-autoencoders-for-paraphrase-detection.pdf
- Is there already an implementation in another framework? (if so, where): no
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**
This would be a new layer.
**Who will benefit with this feature?**
There have been too many requests and issues on the Keras page raised for implementation of min pooling because(unlike custom pooling) it helps extract data from high contrast/brightness images where max-pooling fails.
**Any other info.**
I would like to start by implementing the basic min pooling, after which as mentioned in the 2nd paper above, a dynamic approach towards min pooling can be taken which would further improve the pooling dynamic to adjust to different image regions differently allowing greater retentivity of information.

**Comment by seanpmorgan** [comment]
Hi @Om-Pandey this seems like a reasonable request. Would you mind linking some of the keras issues so we can review what the comments were?

**Comment by Om-Pandey** [comment]
Sure .... check this:
https://github.com/keras-team/keras/issues/2816

**Comment by Om-Pandey** [comment]
@seanpmorgan can I start working on this? please assign me this issue. Also, please review #806 and tell me possible steps as to what to do next ...😄

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
